Source/plrmsg.cpp

```diff
@@ -49,87 +62,69 @@ void plrmsg_delay(bool delay)
 	}
 
 	plrmsgTicks += SDL_GetTicks();
-	_plrmsg *pMsg = plr_msgs;
-	for (int i = 0; i < PMSG_COUNT; i++, pMsg++)
-		pMsg->time += plrmsgTicks;
+	for (PlayerMessage &message : Messages)
+		message.time += plrmsgTicks;
 }
 
-void ErrorPlrMsg(const char *pszMsg)
+void EventPlrMsg(string_view text, UiFlags style)
 {
-	_plrmsg *pMsg = &plr_msgs[plr_msg_slot];
-	plr_msg_slot = (plr_msg_slot + 1) & (PMSG_COUNT - 1);
-	pMsg->player = MAX_PLRS;
-	pMsg->time = SDL_GetTicks();
-	CopyUtf8(pMsg->str, pszMsg, sizeof(pMsg->str));
-}
+	PlayerMessage &message = GetNextMessage();
 
-size_t EventPlrMsg(const char *pszFmt, ...)
-{
-	_plrmsg *pMsg;
-	va_list va;
-
-	va_start(va, pszFmt);
-	pMsg = &plr_msgs[plr_msg_slot];
-	plr_msg_slot = (plr_msg_slot + 1) & (PMSG_COUNT - 1);
-	pMsg->player = MAX_PLRS;
-	pMsg->time = SDL_GetTicks();
-	vsprintf(pMsg->str, pszFmt, va);
-	va_end(va);
-	return strlen(pMsg->str);
+	message.style = style;
+	message.time = SDL_GetTicks();
+	message.text = std::string(text);

```

Comment:
This could also initialise from as a string_view of length 0 to match SendPlrMsg. Not really necessary though, default initialiser makes it `{nullptr,0}`, this would make it `{text.data(),0}` instead.

```suggestion
	message.text = std::string(text);
	message.from = std::string_view(message.text, 0);
```